### DISCUSSION TITLE
Automated Install on Proxmox?

### DISCUSSION THREAD

**Discussion by Just-Insane** [siderolabs/talos#6970] [Q&A]
I am looking to see what would be required to enable a more automated cluster install on Proxmox.

From what I've seen, the options for a more automated install are listed in https://github.com/siderolabs/talos/discussions/4944, and were outlined as:

1. Use a nocloud image instead of the default metal based ISO, this would require building an ISO for use on Proxmox.
2. Take a snapshot (seems irrelevent for Proxmox)
3. Use Packer to create an ISO, which links to some old documentation on using Packer, and appears to create an hcloud image, again irrelevent.

I am using Terraform to deploy the cluster, however, due to not being able to install the qemu-guest-agent, terraform is not able to get the guest IP from DHCP, which means I have to either look in the console manually or check my DHCP server for leases, and then populate machineconfigs with a static IP and then apply those configs.

It seems like the best path forward would be:

Generate an ISO using the nocloud image, which I can then use Proxmox's cloudinit support to populate static IPs on the nodes and generate machine configs.

An alternative would be to create an image that includes qemu-guest-agent prior to packaging, though I feel like that would be against the principals of the project.

Once the cluster is up, I can use https://github.com/crisobal/qemu-guest-agent-talos to install the agent via a daemonset, but that still leaves me with no way of automating machineconfig deployment from a standard ISO.

Maybe I'm missing something obvious that would make this less complicated?

**Comment by alexpdp7** [comment]
When https://github.com/siderolabs/talos/pull/5897 is merged, you should be able to just use host names.

In the meantime, I pull the MAC address from the Proxmox VM file and use that to find the IP from dnsmasq, using this script:

https://github.com/alexpdp7/alexpdp7/blob/master/personal_infra/playbooks/roles/talos/files/get-ip

This playbook is poorly documented, but might serve you as a base:

https://github.com/alexpdp7/alexpdp7/tree/master/personal_infra/playbooks/roles/talos

(Happy to answer questions about that.)

**Reply by Just-Insane** [reply]
I was not aware that that was a planned change, but look forward to it.

I’ll experiment with your script to see if I can get it working in Terraform to pull the current IPs via DHCP and use them for the bootstrapping.
